scripts/rollup/rollup.config.js

```diff
@@ -27,16 +27,16 @@ const { ROLLUP_WATCH: watchMode } = process.env;
 const formats = ['es', 'cjs'];
 
 if (packageName === '@lwc/synthetic-shadow') {
-    // Here we wrap all of synthetic shadow in a check for lwcRuntimeFlags.ENABLE_FORCE_SHADOW_MIGRATE_MODE, so
-    // that synthetic shadow is not loaded at all if the flag is in effect.
+    // Here we wrap all of synthetic shadow in a check for lwcRuntimeFlags.ENABLE_FORCE_SHADOW_MIGRATE_MODE and
+    // lwcRuntimeFlags.DISABLE_SYNTHETIC_SHADOW, so that synthetic shadow is not loaded if either flag is set.
     // Note that lwcRuntimeFlags must be referenced as a pure global, or else string replacement in ESBuild
     // will not work. But we also have to check to make sure that lwcRuntimeFlags is defined, so this
     // `Object.defineProperty` code is copied from @lwc/features itself.
     banner += `
     if (!globalThis.lwcRuntimeFlags) {
       Object.defineProperty(globalThis, 'lwcRuntimeFlags', { value: Object.create(null) });
     }
-    if (!lwcRuntimeFlags.ENABLE_FORCE_SHADOW_MIGRATE_MODE) {
+    if (!lwcRuntimeFlags.ENABLE_FORCE_SHADOW_MIGRATE_MODE && !lwcRuntimeFlags.DISABLE_SYNTHETIC_SHADOW) {
```

Comment:
Not every environment has the runtime flag set before `@lwc/synthetic-shadow` executes, but enough do that it makes sense to just disable synthetic shadow entirely in those cases.

Reply:
Unfortunately there are still apps that don't set flags before LWC bootstraps. Ideally yes we would not need this code, but this is the next-best thing.